docs/README.Windows.md

```diff
@@ -0,0 +1,244 @@
+
+### NSIS install notes
+Default options are fine.
+
+### Visual Studio 2017 install notes
+Start the VS2017 installer and click `Individual components`.
+* Under **Compilers, build tools and runtimes** select
+  * `Msbuild`
+  * `VC++ 2017 v141 toolset (x86,x64)`
+  * `Visual C++ 2017 Redistributable Update`
+  * `Visual C++ compilers and libraries for ARM`
+  * `Visual C++ compilers and libraries for ARM64`
+  * `Visual C++ runtime for UWP`
+  * `Windows Universal CRT SDK`
+* Under **Development activities** select
+  * `Visual Studio C++ core features`
+* Under **SDKs, libraries, and frameworks** select
+  * `Windows 10 SDK (10.0.16299.0) for Desktop C++ [x86 and x64]`
+  * `Windows 10 SDK (10.0.16299.0) for UWP: C++`
+
+Hit `Install`. Yes, it will download and install almost 7GB of stuff.
+
+This is all you need to do a *normal* Kodi build for 32 or 64bit. Building for UWP (Universal Windows Platform) requires the above listed and quite a lot more.
+
+Under `Workloads` select
+ * `Universal Windows Platform development`
+ * `Desktop development with C++`
+
+Hit `Install`. It will download and install an extra 12GB of whatever for a grand total of almost 20GB. Yes, seriously!
+
+**[back to top](#table-of-contents)** | **[back to section top](#2-prerequisites)**
+
+## 3. Get the source code
+Change to your `home` directory:
+```
+cd %userprofile%
+```
+
+Clone Kodi's current master branch:
+```
+git clone https://github.com/xbmc/xbmc kodi
+```
+
+**[back to top](#table-of-contents)**
+
+## 4. Set up the build environment
+To set up the build environment, several scripts must be called.
+
+**WARNING:** The scripts may fail if you have a space in the path to the bat files.
+
+Kodi can be built as either a normal 32bit or 64bit program, UWP 32bit and 64bit and UWP ARM 32bit. Unless there is a reason to prefer 32bit builds, we advise you to build Kodi for 64bit.
+
+**TIP:** Look for comments starting with `Or ...` and only execute the command(s) you need.
+
+Change to the 64bit build directory (**recommended**):
+```
+cd %userprofile%\kodi\tools\buildsteps\windows\x64
+```
+
+Or change to the 32bit build directory:
+```
+cd %userprofile%\kodi\tools\buildsteps\windows\win32
+```
+
+Or change to the UWP 64bit build directory:
+```
+cd %userprofile%\kodi\tools\buildsteps\windows\x64-uwp
+```
+
+Or change to the UWP 32bit build directory:
+```
+cd %userprofile%\kodi\tools\buildsteps\windows\win32-uwp
+```
+
+Or change to the UWP ARM 32bit build directory:
+```
+cd %userprofile%\kodi\tools\buildsteps\windows\arm-uwp
+```
+
+Download dependencies:
+```
+download-dependencies.bat
+```
+**TIP:** Look for the `All formed packages ready!` success message. If you see the message `ERROR: Not all formed packages are ready!`, execute the command again until you see the success message.
+
+Download and setup the build environment for libraries:
+```
+download-msys2.bat
+```
+
+Build FFmpeg:
+```
+make-mingwlibs.bat
```

Comment:
It is worth adding that this takes a rediculously long time, and appears to be doing absolutely nothing for nearly 10 minutes just after the start. You need to be patient! Don't abort because there is nothing visibly happening. For some reason this is much slower on Windows than other platforms.

Reply:
lol, yep. But that started with the latest ffmpeg build, 4.0, iirc. I wrote the guide with the ***old*** ffmpeg build.

Besides, everything takes forever on Windows. ;-)

Reply:
Thanks for doing this useful task @hudokkow 
I have always had problems with building ffmpeg  more than any other part. Now it is just a 10min wait for action, that still needs a warning that it has not crashed. But it used to be worse, once I had it showing progress (I thought) by putting dots on the window so I left it running for 3 hours (my old cpu was slow too). In fact that was all it was doing, putting dots on the screen!!!

Having never built on any other platform I don't realise how slow it is (but I do notice the platform times on jenkins)